ping/ping_output.c

```diff
@@ -0,0 +1,230 @@
+/*
+ * SPDX-License-Identifier: BSD-3-Clause
+ * Copyright (c) 2024-2025 Georg Pfuetzenreuter <mail+ip@georg-pfuetzenreuter.net>
+ */
+
+#include "ping.h"
+#include "stdarg.h"
+
+void ping_print_int(struct ping_rts *rts, char *msg, char *json_key, int json_val) {
+	if (rts->opt_json) {
+		construct_json(rts, PING_JSON_INT, json_key, json_val);
+	} else {
+		printf(msg, json_val);
+	}
+}
+
+void ping_print_uint(struct ping_rts *rts, char *msg, char *json_key, unsigned int json_val) {
+	if (rts->opt_json) {
+		construct_json(rts, PING_JSON_UINT, json_key, json_val);
+	} else {
+		printf(msg, json_val);
+	}
+}
+
+void ping_print_str(struct ping_rts *rts, char *msg, char *json_key, char *json_val) {
+	if (rts->opt_json) {
+		construct_json(rts, PING_JSON_STR, json_key, json_val);
+	} else {
+		printf(msg, json_val);
+	}
+}
+
+inline void ping_print_version(struct ping_rts *rts) {
+	if (rts->opt_json) {
+		construct_json(rts, PING_JSON_STR, "version", PACKAGE_VERSION);
+		print_json_packet(rts);
+	} else {
+		printf(PACKAGE_VERSION);
+		print_config();
+	}
+}
+
+inline void ping_print_error_packet(struct ping_rts *rts, char *address, uint8_t sequence) {
+	if (!rts->opt_json)
+		printf(_("From %s icmp_seq=%u "), address, sequence);
+}
+
+inline void ping_print_error_parse(struct ping_rts *rts, char *reason) {
+	if (!rts->opt_json)
+		printf(_(" parse error (%s)"), reason);
+	construct_json(rts, PING_JSON_STR, "error", reason);
+}
+
+inline void ping_print_error_qtype(struct ping_rts *rts, uint32_t val) {
+	if (!rts->opt_json)
+		printf(_(" unknown qtype(0x%02x)"), val);
+	construct_json(rts, PING_JSON_STR, "error", "unknown qtype");
+}
+
+inline void ping_print_truncated(struct ping_rts *rts) {
+	if (!rts->opt_json) {
+		printf(_(" (truncated)"));
+		if (rts->ni.subject_type == IPUTILS_NI_ICMP6_SUBJ_IPV4)
+			putchar('\n');
+	}
+	construct_json(rts, PING_JSON_STR, "info", "truncated");
+}
+
+inline void ping_print_packet(struct ping_rts *rts) {
+	if (!rts->opt_json) {
```

Comment:
Please use:
```C
if (rts->opt_json)
    return;
```
to avoid whole function code being indented.

Reply:
Adjusted, but I kept the `construct_json()` call inside the condition, as otherwise `opt_json` gets uselessly checked twice if it is false (once in `ping_print_packet()` and once in `construct_json()`).